src/extension.ts

```diff
@@ -648,6 +648,28 @@ export class Extension implements RunHooks {
       await this._queueWatchRun(new this._vscode.TestRunRequest(testItems), 'items');
   }
 
+  private async _showBrowserForRecording(file: string, project: TestProject, connectWsEndpoint: string) {
+    const fileItem = this._testTree.testItemForFile(file);
+    if (!fileItem)
+      throw new Error(`File item not found for ${file}`);
+    if (fileItem.children.size !== 1)
+      throw new Error('expected single test in file');
+
+    const testItems = this._testTree.collectTestsInside(fileItem);
+    const testForProject = testItems.length === 1 ? testItems[0] : testItems.find(t => t.label === project.name);
+    if (!testForProject)
+      throw new Error(`Test item not found for ${project.name}`);
+
+    const request = new this._vscode.TestRunRequest([testForProject], undefined, undefined, false, true);
+    (request as any)[kTestRunOptions] = {
```

Comment:
It seems like you need to check `showBrowser` setting instead, and uncheck it after the recording has finished?